crates/snapbox/src/path.rs

```diff
@@ -181,8 +183,14 @@ impl PathDiff {
                     let expected = crate::Data::read_from(&expected_path, None)
                         .map(|d| d.map_text(crate::utils::normalize_lines))
                         .map_err(Self::Failure)?;
-                    if expected.as_str().is_some() {
-                        actual = actual.try_text().map_text(crate::utils::normalize_lines);
+
+                    if expected.format() != DataFormat::Binary {
+                        match actual.coerce(expected.format()) {
+                            Ok(d) => {
+                                actual = d.map_text(crate::utils::normalize_lines);
+                            }
+                            Err(e) => return Err(Self::Failure(e)),
```

Comment:
I used `Failure` here since none of the other error types gave a true description of what was going on. If needed a new error (`DataFormatMismatch` or similar) could be added if needed.